feed/models.py

```diff
@@ -0,0 +1,61 @@
+from django.db import models
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+
+from django.contrib.auth.models import User
+from course.models import TeacherCourses, StudentCourses
+
+
+class PostManager(models.Manager):
+    @staticmethod
+    def get_posts_for_user(user: User):
+        relevant_users = StudentCourses.objects.filter(
+            Q(Q(student_id=user) & Q(status="Confirmed") | Q(teacher_course_id__teacher_id=user))
```

Comment:
Interesting use of Q!

Reply:
What is the best practice for filtering based on a "nested" (not sure about the exact term) boolean expression?

Reply:
I wasn't sarcastic, it is a nice usage. but if you found an error and fixed it, that's good :)